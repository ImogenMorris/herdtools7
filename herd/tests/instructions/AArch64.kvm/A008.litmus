AArch64 A008
variant=fatal
(**
 * This tests raises the warning
 * "Fault inside a fault handler. Legal outcomes may be missing"
 * However, the handler "P0.F" triggers the fault only if
 * a CoRR violation occurs, which  any model will forbid.
 *Future fix may get rid of the warning.
**)
{
0:X0=x; 0:X2=z;
1:X0=x;
[PTE(z)]=(valid:0);
}
 P0          | P1          |   P0.F      ;
 LDR W1,[X0] | MOV W1,#1   | CBZ W1,L0   ;
 LDR W3,[X0] | STR W1,[X0] | CBNZ W3,L0  ;
 LDR W5,[X2] |             | LDR W7,[X2] ;
             |             |L0:          ;             
locations [0:X1;0:X3;]
exists 0:X1=1 /\ 0:X3=0
