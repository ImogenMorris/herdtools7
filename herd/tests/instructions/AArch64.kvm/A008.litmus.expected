Test A008 Allowed
States 3
0:X1=0; 0:X3=0;
0:X1=0; 0:X3=1;
0:X1=1; 0:X3=1;
No
Witnesses
Positive: 0 Negative: 3
Condition exists (0:X1=1 /\ 0:X3=0)
Observation A008 Never 0 3
Hash=47c0d5abe4c93f43062380b5cfad18be

